src/image/PhotoCalib.cc

```diff
@@ -293,6 +293,85 @@ MaskedImage<float> PhotoCalib::calibrateImage(MaskedImage<float> const &maskedIm
     return result;
 }
 
+afw::table::SourceCatalog PhotoCalib::calibrateCatalog(afw::table::SourceCatalog const &catalog,
+                                                       std::vector<std::string> const &instFluxFields) const {
+    auto const &inSchema = catalog.getSchema();
+    afw::table::SchemaMapper mapper(inSchema, true);  // true: share the alias map
+    mapper.addMinimalSchema(inSchema);
+
+    std::vector<table::Key<double>> instFluxKeys;
+    instFluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> instFluxErrKeys;
+    instFluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxKeys;
+    fluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxErrKeys;
+    fluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magKeys;
+    magKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magErrKeys;
+    magErrKeys.reserve(instFluxFields.size());
+    for (auto const &field : instFluxFields) {
+        instFluxKeys.push_back(inSchema.find<double>(field + "_instFlux").key);
+        try {
+            instFluxErrKeys.push_back(inSchema.find<double>(field + "_instFluxErr").key);
+        } catch (pex::exceptions::NotFoundError) {
```

Comment:
Even though you're not doing anything with this exception, I think you still want to catch it by reference, i.e. `catch (pex::exceptions::NotFoundError&)`.  I'm not positive there would be any ill effects from this as is, and I don't even remember in detail what the ill effects of catching by value are, but it's safest to just always catch by reference.